# Exercise

You are given height_weight.csv file which contains heights and weights of 1000 people. Dataset is taken from here,
https://www.kaggle.com/mustafaali96/weight-height

You need to do this,

(1) Load this csv in pandas dataframe and first plot histograms for height and weight parameters

(2) Using IQR detect weight outliers and print them

(3) Using IQR, detect height outliers and print them

In [1]:
import pandas as pd
df = pd.read_csv("height_weight.csv")
df.head()

gender     height      weight
0   Male  73.847017  241.893563
1   Male  68.781904  162.310473
2   Male  74.110105  212.740856
3   Male  71.730978  220.042470
4   Male  69.881796  206.349801

In [2]:
df.describe()

height        weight
count  10000.000000  10000.000000
mean      66.367560    161.440357
std        3.847528     32.108439
min       54.263133     64.700127
25%       63.505620    135.818051
50%       66.318070    161.212928
75%       69.174262    187.169525
max       78.998742    269.989699

In [3]:
height_Q1 = df["height"].quantile(0.25)

height_Q3 = df["height"].quantile(0.75)

print(height_Q1)
print(height_Q3)

63.505620481218955
69.1742617268347


In [4]:
height_IQR = height_Q3 - height_Q1

height_IQR

5.668641245615746

In [6]:
height_min_threshold = height_Q1 - 1.5*height_IQR
height_max_threshold = height_Q3 + 1.5*height_IQR

height_min_threshold, height_max_threshold

(55.00265861279534, 77.67722359525831)

In [7]:
df_height_outliers = df[(df["height"]>height_max_threshold) | (df["height"]<height_min_threshold)]

df_height_outliers

gender     height      weight
994     Male  78.095867  255.690835
1317    Male  78.462053  227.342565
2014    Male  78.998742  269.989699
3285    Male  78.528210  253.889004
3757    Male  78.621374  245.733783
6624  Female  54.616858   71.393749
7294  Female  54.873728   78.606670
9285  Female  54.263133   64.700127

In [8]:
df_height_no_outliers = df[(df["height"]<height_max_threshold) & (df["height"]>height_min_threshold)]

df_height_no_outliers.shape

(9992, 3)

We removed 8 height outliers

In [9]:
weight_Q1 = df["weight"].quantile(0.25)

weight_Q3 = df["weight"].quantile(0.75)

print(weight_Q1)
print(weight_Q3)

135.8180513055015
187.16952486868348


In [10]:
weight_IQR = weight_Q3 - weight_Q1

weight_IQR

51.35147356318197

In [12]:
weight_min_threshold = weight_Q1 - 1.5*weight_IQR
weight_max_threshold = weight_Q3 + 1.5*weight_IQR

weight_min_threshold, weight_max_threshold

(58.79084096072856, 264.19673521345646)

In [13]:
df_weight_outliers = df[(df["weight"]>weight_max_threshold) | (df["weight"]<weight_min_threshold)]

df_weight_outliers

gender     height      weight
2014   Male  78.998742  269.989699

This person is an outlier in both height an weight

In [14]:
df_no_outliers = df_height_no_outliers[(df["weight"]<weight_max_threshold) & (df["weight"]>weight_min_threshold)]

df_no_outliers.shape

/var/folders/r3/_d0_r97169g0lxpyj8681vgw0000gn/T/ipykernel_36662/1551968992.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  df_no_outliers = df_height_no_outliers[(df["weight"]<weight_max_threshold) & (df["weight"]>weight_min_threshold)]


(9992, 3)

We already removed the weight outlier, so in total 9992 good rows